source/agora/common/EnrollmentManager.d

```diff
@@ -125,6 +136,15 @@ public class EnrollmentManager
     {
         static ubyte[] buffer;
 
+        // check validity of the enrollment data
+        if (auto reason = isInvalidEnrollmentReason(this.ledger.getBlockHeight() + 1,
+            enroll, this.utxo_set.getUTXOFinder()))
+        {
+            this.logMessage("Invalid enrollment data, Reason: " ~ reason,
```

Comment:
This will allocate a new string every time enrollment fails validation. See this comment here: https://github.com/bpfkorea/agora/pull/465#discussion_r364074502

Reply:
The logMessage function should be enhanced for general usage also. In the PR for the enhancement, I will change the code with your comment.